### ISSUE TITLE
Training with custom data gets stuck on the first iteration.

### ISSUE THREAD

**Issue by SamiurRahman1** [siavash-khodadadeh/MetaLearning-TF2.0#15]
When i am trying to train a model with my custom data, it is stuck with the following output:

```
No previous checkpoint found!
0it [00:00, ?it/s]
```

It stays like this until i interrupt the training. I can also see that the GPU is in use but i don't see anything else happening. Is it normal? As far as i know, i should be able to see some logs like accuracy, loss, epoch count etc.

**Comment by siavash-khodadadeh** [comment]
Yes, you should be able to see some logs here. Can you please check if a folder is created in [maml](https://github.com/siavash-khodadadeh/MetaLearning-TF2.0/tree/master/models/maml) directory for your logs?

**Comment by SamiurRahman1** [comment]
a folder is actually created in the master folder as i moved my script in the master folder before i ran it. Of course i would love to share my code and dataset.

**Comment by SamiurRahman1** [comment]
i have created a fork and uploaded my code and here is a link for the dataset and logfiles. Thank you again for your help.

Dataset: https://drive.google.com/file/d/1l37mCGycof3qI58gvjfDJ5tCE_cISEGz/view?usp=sharing

log file: https://drive.google.com/file/d/1yMrQlwn9AqaVWvOBuBt-jyCYPPAEsn27/view?usp=sharing

**Comment by siavash-khodadadeh** [comment]
Thanks! Please share the link to the fork with me. I will look into it soon, however, I am a little busy with some other projects now so it might take a couple of days before I get back to you.

**Comment by SamiurRahman1** [comment]
here is the link of the repo https://github.com/SamiurRahman1/MetaLearning-TF2.0

**Comment by siavash-khodadadeh** [comment]
Can you please point me to the python files you added?

**Comment by SamiurRahman1** [comment]
Sorry, not very familiar with github generally. Here are the script links:
https://drive.google.com/file/d/1183N5iO-fZugJ6_lXetZuno53W-uxrLS/view?usp=sharing, https://drive.google.com/file/d/1XelBlei1WLE3o5kOnzho4jupmpQmGng9/view?usp=sharing

**Comment by SamiurRahman1** [comment]
Hello, did you have some time to look at the code?

**Comment by SamiurRahman1** [comment]
Hi, i am still waiting for reply on this if possible.

**Comment by siavash-khodadadeh** [comment]
Hello, 

I have been a little bit busy. Unfortunately, I cannot check this from python files you shared on google drive since it is hard to track changes. Please put everything on Github and I can check out to that particular branch and debug it.

Thank you very much,
Siavash

**Comment by SamiurRahman1** [comment]
Hi, uploading or creating a branch is disabled for your repo, for obvious reasons. I created a pull request and uploaded the files. Maybe you can find them there? If not, could you please tell me exactly how i should upload them?

**Comment by siavash-khodadadeh** [comment]
I looked at the codes. It seems to me that your dataset has only 6 classes? Is that correct? In that case, do you want to do meta-learning on it or do you want to just use it for test?

**Comment by SamiurRahman1** [comment]
Hi, thanks for your reply. I'm trying to train a model with the dataset. I'll try your suggestion and get back to you.

**Comment by SamiurRahman1** [comment]
Hi, so i tried running the training with meta-batch-size=1 but unfortunately it still gets stuck. Is there anything else i need to change if i want to train the model with only 6 classes?

**Comment by siavash-khodadadeh** [comment]
Okay, I see in the dataset class there are 4 classes for training and 2 classes for validation. In this case, there is no way to generate 5-way tasks during training because there are only 4 classes for training. Can you please try using all 6 classes for training and all 6 classes as well for validation and test to just see if that is the problem? Also, you can set n=4 instead of 5 with meta-batch-size=1, but again since your validation has just two classes, I think you might get the same problem.

**Comment by SamiurRahman1** [comment]
When i set `n=4 , num_train_classes=6, num_val_classes=6`, it throws the following error:
```
ValueError: in user code:

    /data/yali/sam/Project/MetaLearning-TF2.0-master/models/base_model.py:254 meta_train_loop  *
        task_final_acc, task_final_loss = self.get_losses_of_tasks_batch(method='train')(
    /data/yali/sam/Project/MetaLearning-TF2.0-master/models/maml/maml.py:237 inner_train_loop  *
        self.create_meta_model(self.updated_models[0], self.model, gradients)
    /data/yali/sam/Project/MetaLearning-TF2.0-master/models/maml/maml.py:133 create_meta_model  *
        model_layer = model_layer.get_layer(layer_name)
    /home/lili/miniconda3/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py:2398 get_layer  **
        raise ValueError('No such layer: ' + name + '.')

    ValueError: No such layer: simple_model.

```

**Comment by siavash-khodadadeh** [comment]
I do not think you should set num_train_classes and num_val_classes to 6 because that means you have at least 12 classes and the others are for the test. Can you please make sure your function 
def get_train_val_test_folders(self) -> Tuple:
returns the same classes. Let me try to write it here:

```
def get_train_val_test_folders(self) -> Tuple:
        damageTypes = list()
        myDir = "/data/yali/sam/Project/MetaLearning-TF2.0-master/data/Family/"
        for item in os.listdir(myDir):
            damageTypes.append(item)

        #print(damageTypes)
        damageImg = list()

        for damage in damageTypes:
            damageImg.append(os.path.join(myDir+damage))
        damageImg.sort()

        num_train_classes = self.num_train_classes
        num_val_classes = self.num_val_classes

        random.shuffle(damageImg)
        train_chars = damageImg
        val_chars = damageImg
        test_chars = damageImg
        #print(train_chars)


        train_classes = {char: [os.path.join(char, instance) for instance in os.listdir(char)] for char in train_chars}
        val_classes = {char: [os.path.join(char, instance) for instance in os.listdir(char)] for char in val_chars}
        test_classes = {char: [os.path.join(char, instance) for instance in os.listdir(char)] for char in test_chars}
        return train_classes, val_classes, test_classes
```

**Comment by SamiurRahman1** [comment]
I made the changes to the function as you suggested, set `n=4 , num_train_classes=3, num_val_classes=3`. Getting the same error unfortunately.

**Comment by siavash-khodadadeh** [comment]
Can you run maml_omniglot?

**Comment by SamiurRahman1** [comment]
with the omniglot dataset you mean?

**Comment by SamiurRahman1** [comment]
hmm, interestingly, i get the same error when i run maml_omniglot.py

**Comment by siavash-khodadadeh** [comment]
It seems to be something from TF version. What is your TensorFlow version?

**Comment by SamiurRahman1** [comment]
my tf version is 2.3.1

**Comment by WuJi1** [comment]
I encounter the same problem when the TF version is 2.3.1. 
Has this error been solved?

**Comment by siavash-khodadadeh** [comment]
The code currently works with TF version 2.2.0-rc2.
